homeassistant/components/switchbee/__init__.py

```diff
@@ -25,17 +25,28 @@
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up SwitchBee Smart Home from a config entry."""
+
     hass.data.setdefault(DOMAIN, {})
     central_unit = entry.data[CONF_HOST]
     user = entry.data[CONF_USERNAME]
     password = entry.data[CONF_PASSWORD]
     websession = async_get_clientsession(hass, verify_ssl=False)
-    api = CentralUnitPolling(central_unit, user, password, websession)
+    api: CentralUnitPolling | CentralUnitWsRPC = CentralUnitPolling(
+        central_unit, user, password, websession
+    )
+
+    # First try to connect and fetch the version
     try:
         await api.connect()
     except SwitchBeeError as exp:
         raise ConfigEntryNotReady("Failed to connect to the Central Unit") from exp
 
+    # Check if websocket version
+    assert isinstance(api.version, str)
+    if "1.4.6" in api.version:
```

Comment:
This looks a bit brittle. Can you move this into the library instead and make it check for >= 1.4.6 instead?

You can use awesomeversion to compare versions

Reply:
[jafar-atili/pySwitchbee@`1.7.11...1.7.13`#diff-01ddb5f75b](https://github.com/jafar-atili/pySwitchbee/compare/1.7.11...1.7.13#diff-01ddb5f75ba601610176ee48e4b3f0d9e44a988e6f88bda4392a3791a94d7287R23)

I think is going to break when version 1.4.7 is released

Reply:
I am working with SwitchBee to understand the way they number their version.

Today, WS version is 1.4.6(1) and the older (polling) were 1.4.X(X)


Once they reply I'll fix it permanently (should be today or tomorrow)

Reply:
> This looks a bit brittle. Can you move this into the library instead and make it check for >= 1.4.6 instead?
> 
> 
> 
> You can use awesomeversion to compare versions

I created a new class that parses the version correctly

Reply:
It looks like your version check will return `False` for `5.0`

[jafar-atili/pySwitchbee@`1.7.3...1.7.16`#diff-01ddb5f75b](https://github.com/jafar-atili/pySwitchbee/compare/1.7.3...1.7.16#diff-01ddb5f75ba601610176ee48e4b3f0d9e44a988e6f88bda4392a3791a94d7287R23)

Reply:
Fixed in 1.7.17, now using `packaging.version` to compare versions

Reply:
That should work 👍  thanks

It would be great if you would switch that to https://github.com/ludeeus/awesomeversion as suggested in https://github.com/home-assistant/core/pull/84442#discussion_r1055972486 so we don't have to load `packaging` at runtime since `awesomeversion` is already available and used in HA

Reply:
Done